### PR TITLE
Bazel x86 Build

### PR DESCRIPTION
**Pull request by dhawkes**
Working bazel build. This invovled the following changes:

1. Added in bazel BUILD and WORKSPACE files along with a .bazelrc file that includes the linker and CXXOPTS flags necessary to build the binary. This includes a bunch of optimization flags as SAT (apparently) needs to be able to run very efficiently to be effective.
2. Made a minor code change to different built in function since we use GCC with bazel instead of clang (which the original SAT uses).

Tested:

```
dthawkes@dthawkes:~/ocp-diag-sat$ bazel-docker
Starting local Bazel server and connecting to it...
INFO: Analyzed 2 targets (15 packages loaded, 78 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 2.584s, Critical Path: 0.05s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action

dthawkes@dthawkes:~/ocp-diag-sat$ ./bazel-bin/src/sat_bin
2023/03/21-22:08:22(UTC) Log: Commandline - ./bazel-bin/src/sat_bin
2023/03/21-22:08:22(UTC) Stats: SAT revision 1.0.9_autoconf, 64 bit binary
2023/03/21-22:08:22(UTC) Log: dthawkes @ dthawkes.c.googlers.com on Fri Feb  3 00:36:59 UTC 2023 from open source release
2023/03/21-22:08:22(UTC) Log: 1 nodes, 64 cpus.
2023/03/21-22:08:22(UTC) Log: Defaulting to 64 copy threads
2023/03/21-22:08:22(UTC) Log: Total 515961 MB. Free 493690 MB. Hugepages 0 MB. Targeting 489971 MB (94%)
2023/03/21-22:08:22(UTC) Log: Prefer plain malloc memory allocation.
2023/03/21-22:08:22(UTC) Log: Using mmap() allocation at 0x7e8fc4d00000.
2023/03/21-22:08:22(UTC) Stats: Starting SAT, 489971M, 20 seconds
2023/03/21-22:08:42(UTC) Log: Region mask: 0x1
2023/03/21-22:08:52(UTC) Log: Seconds remaining: 10
2023/03/21-22:09:06(UTC) Stats: Found 0 hardware incidents
2023/03/21-22:09:06(UTC) Stats: Completed: 2520142.00M in 20.13s 125171.87MB/s, with 0 hardware incidents, 0 errors
2023/03/21-22:09:06(UTC) Stats: Memory Copy: 2520142.00M at 125346.40MB/s
2023/03/21-22:09:06(UTC) Stats: File Copy: 0.00M at 0.00MB/s
2023/03/21-22:09:06(UTC) Stats: Net Copy: 0.00M at 0.00MB/s
2023/03/21-22:09:06(UTC) Stats: Data Check: 0.00M at 0.00MB/s
2023/03/21-22:09:06(UTC) Stats: Invert Data: 0.00M at 0.00MB/s
2023/03/21-22:09:06(UTC) Stats: Disk: 0.00M at 0.00MB/s
2023/03/21-22:09:06(UTC)
2023/03/21-22:09:06(UTC) Status: PASS - please verify no corrected errors
2023/03/21-22:09:06(UTC)
```

### CODE REVIEW

**Comment by dhawkes** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/4" target="_blank">Graphite</a> 
    * **PR #5** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/5" target="_blank">Graphite</a>  👈
      * **PR #2** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/2" target="_blank">Graphite</a> 
        * **PR #6** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/6" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/5?utm_source=stack-comment).

**Comment by dhawkes** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/4" target="_blank">Graphite</a> 
    * **PR #5** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/5" target="_blank">Graphite</a>  👈
      * **PR #2** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/2" target="_blank">Graphite</a> 
        * **PR #6** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/6" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/5?utm_source=stack-comment).

File: .bazelrc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Include system libraries
```
Comment (kimmidi) [inline .bazelrc:1]: I was expecting to see a Makefile from which you source these options, and the Makefile being deleted in this PR. I don't find any Makefile in the repo though ?

**Review by dhawkes — COMMENTED** [review]
Also worth noting that the flags aren't all the same. The optimization flags I just copied over, but the other ones I figured with googling and trial and error.

File: .bazelrc (hunk 1 above)
Comment (kimmidi) [reply .bazelrc:1]: OK. I was hoping to see that https://github.com/opencomputeproject/ocp-diag-sat/pull/3/files. Would help to understand these changes.

**Review by kimmidi — APPROVED** [review]

**Pull request merged**